lib/jxl/encode.cc

```diff
@@ -42,8 +42,23 @@ JxlEncoderStatus JxlEncoderStruct::RefillOutputByteQueue() {
       std::move(input_frame_queue[0]);
   input_frame_queue.erase(input_frame_queue.begin());
 
-  // TODO(zond): If the frame queue is empty and the input_closed is true,
-  // then mark this frame as the last.
+  // If the frame queue is empty and input_closed is true,
+  // mark this frame as the last.
+  jxl::FrameInfo frame_info;
+  if (input_closed && input_frame_queue.empty()) {
+    frame_info.is_last = true;
+  } else {
+    frame_info.is_last = false;
+  }
+
+  // don't encode more than one non-animation frame,
```

Comment:
Is this part still needed? It's of course not really useful yet to do non-animation multi-frame when we can't set blend modes / cropped frames yet, but it should work I think (it will just be full frames using whatever the default blend mode is, which is probably not useful).

Reply:
I don't know if it's needed.  But it prevents users from creating files with frames they cannot recover yet.

Reply:
Fair enough, but maybe then return `JXL_ENC_NOT_SUPPORTED` in cases where someone is trying to produce a multi-layer file, instead of silently forcing the input to be closed after the first layer?

We don't want to have applications rely on behavior (implicit closing of input in case of !have_animation) that we know will change at some point.

Reply:
Okay.  Will make that change.